### DISCUSSION TITLE
Anticipated zod v4 release date?

### DISCUSSION THREAD

**Discussion by dieseldjango** [colinhacks/zod#3863] [Q&A]
I've seen the great news about Colin's [fellowship](https://zod.dev/blog/clerk-fellowship) to work on zod 4. On my current project we made a big bet on zod for runtime validation, and it worked great at first but we're now facing unacceptable performance dealing with large object graphs. If a v4 release with major performance improvements is imminent I can make the argument to wait until it's available or even to provide some dev assistance in getting it over the finish line if that would be helpful. Otherwise, it's looking like we will need to look at an alternative.

Can you provide any idea of just how close v4 is to being ready?

**Comment by maurer2** [comment]
Hello, the first beta of Zod4 is supposed to be released soon according to this tweet: https://x.com/colinhacks/status/1882916290404458516.

Cheers
